### PR TITLE
add scripts to make a zip file with modules and test it

### PR DESCRIPTION
**Pull request by matthiasdiener**
Addresses https://github.com/illinois-ceesd/planning/issues/26.

### CODE REVIEW

File: testzip.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python3
```
Comment (matthiasdiener) [inline testzip.py:1]: not sure if we should keep this file.

File: makezip.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -o errexit -o nounset
```
Comment (inducer) [inline makezip.sh:3]: Could you  make this a variant of the normal install flow?

File: makezip.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+echo "=== Preparing path file of '$MY_PYTHON'"
+echo "=== for importing modules from '$zipfile'"
+echo
+
+sitefile="$($MY_PYTHON -c 'import site; print(site.getsitepackages()[0])')/emirge.pth"
+
+echo "$zipfile" > "$sitefile"
+
+echo "=== Done. Make sure to uninstall other copies of the emirge modules and"
```
Comment (inducer) [inline makezip.sh:30]: > Make sure to uninstall other copies of the emirge modules

I think this comment is kind of unhelpful. I wouldn't trust myself to do this correctly. How will users accomplish this?

File: testzip.py (hunk 1 above)
Comment (inducer) [reply testzip.py:1]: Could you add a CI job that tests the zip install flow?

Comment (matthiasdiener) [reply testzip.py:1]: Added a simple CI job that doesn't use this script. I'll remove this script unless there is interest in keeping it.

File: makezip.sh (hunk 3 above)
Comment (matthiasdiener) [reply makezip.sh]: I added more details to the echo here. The reason is that if a module is also installed via the normal `pip` commands, that pipp'ed module seems to have preference over the modules.zip module.

File: makezip.sh (hunk 2 above)
Comment (matthiasdiener) [reply makezip.sh:3]: I added the --modules parameter to the main install.sh script.

File: testzip.py (hunk 1 above)
Comment (inducer) [reply testzip.py:1]: I agree, this can go.

File: makezip.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -o errexit -o nounset
+
+MY_MODULES=$(git submodule status | awk '{print $2}' | tr '\n' ' ')
```
Comment (inducer) [inline makezip.sh:5]: What if we add a non-python submodule? You could check for the presence of `setup.py` in the subdirectory.

File: install.sh [hunk 5]
```diff
@@ -11,13 +11,19 @@ echo
...
+opt_modules=0
+
 while [[ $# -gt 0 ]]; do
   arg=$1
   shift
   case $arg in
     --prefix=*)
         conda_prefix=${arg#*=}
         ;;
+    --modules)
```
Comment (inducer) [inline install.sh:24]: Could you document this?

File: makezip.sh (hunk 4 above)
Comment (matthiasdiener) [reply makezip.sh:5]: Done.

File: install.sh (hunk 5 above)
Comment (matthiasdiener) [reply install.sh:33]: Added --help parameter and comments.

**Comment by inducer** [comment]
Thanks for working on this. We should also document somewhere how to use this, so that folks trying to run on big, distributed machines aren't confused.

**Comment by matthiasdiener** [comment]
> Thanks for working on this. We should also document somewhere how to use this, so that folks trying to run on big, distributed machines aren't confused.

Maybe in a (emirge/mirgecom) wiki page? Or add it in a (README) file inside emirge/mirgecom itself?

**Comment by inducer** [comment]
> Maybe in a (emirge/mirgecom) wiki page? Or add it in a (README) file inside emirge/mirgecom itself?

Maybe the README? It's just a sentence IMO, and the README is more
likely to stay up to date.

**Pull request merged**